app/controllers/register/company/AddCompanyDirectorsController.scala

```diff
@@ -78,4 +84,26 @@ class AddCompanyDirectorsController @Inject() (
         )
       }
   }
+
+  private def directorsWithFlag(
+                                 withFlag: Seq[(DirectorDetails, Boolean)],
+                                 directors: Seq[DirectorDetails],
+                                 flags: Seq[Boolean]
+                               ): Seq[(DirectorDetails, Boolean)] = {
+
+    if(directors.isEmpty){
+      withFlag
+    } else if (flags.isEmpty) {
+      val addWithFlag: (DirectorDetails, Boolean) = (directors.head, false)
+      directorsWithFlag(withFlag :+ addWithFlag, directors.tail, Seq.empty)
+    } else {
+      val addWithFlag: (DirectorDetails, Boolean) = (directors.head, flags.head)
+      directorsWithFlag(withFlag :+ addWithFlag, directors.tail, flags.tail)
+    }
+
+  }
+
+  private def disableSubmission(directorsWithFlag: Seq[(DirectorDetails, Boolean)]): Boolean = directorsWithFlag.foldLeft(true){ (x, y) =>
+    !y._2
```

Comment:
we don't really need to declare `x`, maybe it'd be good if we renamed `y` to `isDeleted` or something like that -- just for readability.

Reply:
Done